content/guides/compute.md

```diff
@@ -4,3 +4,331 @@ title: Compute Engine documentation | ARGO
+- which status type is considered as default in unknown circumstances
+- a list of available operations between statuses expressed in the form of truth tables
+ 
+The available status states in the metric enviroment are expressed in the "states" list as described below 
+```
+"states":
+  [
+    "OK",
+    "WARNING",
+    "UNKNOWN",
+    "MISSING",
+    "CRITICAL",
+    "DOWNTIME"
+  ]
+```
+
+Based on the availabe states declared in the "states" list are then declared default missing,unknown and downtime states for eg:
+
+```
+ "default_down": "DOWNTIME",
+ "default_missing": "MISSING",
+ "default_unknown": "UNKNOWN",
+```
+
+The available operations are declared in the operations list using truth tables as follows:
+
+```
+"operations":
+  {
+    "AND":[]
+    "OR":[]
+  }
+```
+Each operation consists of a json array used to describe a truth table. For example expanding the above AND operation we see that it consists of a list of dictionary elements
+
+```
+"operations":
+  {
+    "AND":
+    [
+      { "A":"OK",       "B":"OK",       "X":"OK"       },
+      { "A":"OK",       "B":"WARNING",  "X":"WARNING"  },
+      { "A":"OK",       "B":"UNKNOWN",  "X":"UNKNOWN"  },
+      ...
+      ...
+      ...
+     ]
+```
+Each element of the json array describes a row of the truth table for eg:  
+` { "A":"OK", "B":"WARNING", "X":"WARNING"}`
+declares that in an algorithmic AND operation between two status states of *OK* and *WARNING* the result is *WARNING*
+
+In the ops file the user is able to declare any number of availabe monitoring states and any number of available custom operations on those states. The compute-engine picks up this information as a sync file for a job and creates in memory the corresponding truth tables.
+
+##### Config File (per tenant/ per job)
+A job config file is a json file that contains specific information needed during the job run such as grouping parameters, the name of the availability profile used and many more.
+
+The file name template of each job config file is the following:  
+`{TENANT_NAME}_{JOB_ID}_cfg.json`
+
+If the tenant's name is T1 and the jobs name is JobA then the filename of the config file must be:
+`T1_JobA_cfg.json`
+
+###### Contents of an job config file
+The configuration file of the job contains mandatory and optional fields with rich information describing the paramaters of the specific job. Some important fields are:
+```
+"tenant":"tenant_name"`
+"job":"job_name",
+"aprofile":"availability_profile_name",
+"egroup":"endpoint_group_type_name",
+"ggroup":"group_of_group_type_name",
+"weight":"weight_factor_type_name",
+```
+In the above snippet are declared the name of the tenant, the name of the job, the name of the specific availability profile used in the job. Also the type of endpoint grouping that will be used is declared here also and the type of upper hierarchical grouping. Also if available here is declared the type of weight factor used for upper level a/r aggregations
```

Comment:
Do we define anywhere what "endpoint grouping" and "hierarchical grouping" are? What are the available types?